spec/ibandit/iban_spec.rb

```diff
@@ -353,6 +353,109 @@
                                      branch_code: "is invalid")
       end
     end
+
+    context "when the IBAN was created with local details for New Zealand" do
+      let(:arg) do
+        {
+          country_code: "NZ",
+          bank_code: "11",
+          branch_code: "2222",
+          account_number: account_number,
+        }
+      end
+      context "with a 3 digit account number suffix" do
+        let(:account_number) { "3333333-044" }
+
+        its(:country_code) { is_expected.to eq("NZ") }
+        its(:bank_code) { is_expected.to eq("11") }
+        its(:branch_code) { is_expected.to eq("2222") }
+        its(:account_number) { is_expected.to eq("3333333044") }
+        its(:swift_bank_code) { is_expected.to eq("11") }
+        its(:swift_branch_code) { is_expected.to eq("2222") }
+        its(:swift_account_number) { is_expected.to eq("3333333044") }
+        its(:swift_national_id) { is_expected.to eq("112222") }
+        its(:iban) { is_expected.to be_nil }
+        its(:pseudo_iban) { is_expected.to eq("NZZZ1122223333333044") }
+        its(:valid?) { is_expected.to eq(true) }
+        its(:to_s) { is_expected.to eq("") }
+
+        context "with a wrongly formatted account number" do
+          # 3-digit NZ account number suffixes must always start with 0
+          let(:account_number) { "3333333-944" }
```

Comment:
Why is this invalid? Three digit suffixes are valid in NZ.

https://en.wikipedia.org/wiki/New_Zealand_bank_account_number

> S are digits of the suffix (2 or 3 digits). Where a bank displays the suffix as two digits, a leading zero is added to pad the suffix to three digits

This doesn't make three digits invalid, but instead means two digits have to be added by one to make three rather than rejecting them entirely.

Reply:
Actually, this comes from the same wikipedia entry you've linked in the `Account Number Suffi` section

<img width="637" alt="NZD Account Number Suffix" src="https://user-images.githubusercontent.com/2794895/43005880-0ead2f90-8c2c-11e8-817c-18f77bc60718.png">

Reply:
To my eyes, that means something different with the context. More like:

> Any bank that has a two digit suffix, also has a three digit suffix which is created padding a two digit suffix from the left

Which is different to "no bank has a genuine three digit suffix".